app/services/ui_notifications/report_finished.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module UINotifications
+  class ReportFinished < UINotifications::Base
+    include Rails.application.routes.url_helpers
+
+    def initialize(composer, job_id)
+      @composer = composer
+      @job_id = job_id
+      super(User.current)
+    end
+
+    private
+
+    def create
+      add_notification
+    end
+
+    def update_notifications
+      blueprint.notifications.
+          where(subject: subject).
+          update_all(expired_at: blueprint.expired_at)
+    end
+
+    def add_notification
+      Notification.create!(
+        initiator: initiator,
+        audience: ::Notification::AUDIENCE_USER,
+        subject: subject,
+        message: N_('Report "%s" is ready to download') % @composer.template.name,
+        notification_blueprint: blueprint,
+        links: [
+          {
+            title: N_('Download Report'),
+            href: report_data_report_template_path(subject, job_id: @job_id),
```

Comment:
This link does not work, instead of 
```
https://foreman.example.tst/templates/report_templates/175-Host%20statuses/report_data?job_id=2ca50401-bc38-4b77-a0c3-39f3de7c8abc
```

it leads to

```
https://foreman.example.tst/templates/report_templates/4-admin/report_data?job_id=2ca50401-bc38-4b77-a0c3-39f3de7c8abc
```

I assume the subject is wrong...